Doc 1371/update merchant documentation for the new onboarding flow (#453)
swan-io/docs: docs/topics/merchants/guide-missing-information.mdx

---
title: Providing missing information
---

# Providing missing information

> If some information is missing before a review can be finalized, your merchant payment method will have the status `WaitingForInformation`.

Reviewer: This is missing what the purpose of this article is i.e. "how to provide missing information."

  merchantPaymentMethod(id: "$MERCHANT_PAYMENT_METHOD_ID") {
    id
    statusInfo {
      __typename
      status
      
      ... on WaitingForInformationMerchantPaymentMethodStatusInfo {
        verificationRequirements {
          ... on MerchantRequirements {
            fields
          }
          ... on SupportingDocumentRequirements {
            supportingDocumentPurposes
          }
          ... on AdditionalInformationRequirements {
            additionalInformation
          }
        }
      }
    }
  }
}
```

### Payload {#payload-requirements}

```json {} showLineNumbers
{
  "data": {
    "merchantPaymentMethod": {
      "id": "$MERCHANT_PAYMENT_METHOD_ID",
      "statusInfo": {
        "__typename": "WaitingForInformationMerchantPaymentMethodStatusInfo",
        "status": "WaitingForInformation",
        "verificationRequirements": [
          {
            "fields": [
              "MerchantWebsite"
            ]
          },
          {
            "supportingDocumentPurposes": [
              "CompanyTreasury"
            ]
          }
        ]
      }
    }
  }
}
```

## Step 2: Check if a project is concerned by additional information {#step-2}

> Some projects are concerned by additional information.

Reviewer: ```suggestion
Some projects require additional information.
```

            ... on DefaultAdditionalInformation {
              type
              value
            }
          }
        }
      }
    }
    ... on Rejection {
      message
    }
    ... on ValidationRejection {
      fields {
        path
        code
        message
      }
    }
  }
}
```

### Payload {#payload-update}

The new provided values appear under `requestMerchantProfileUpdate`. They will only be reflected on the merchant profile after the request has been reviewed and approved.

```json {} showLineNumbers
{
  "data": {
    "requestMerchantProfileUpdate": {
      "__typename": "RequestMerchantProfileUpdateSuccessPayload",
      "merchantProfile": {
        "id": "$MERCHANT_PROFILE_ID",
        "merchantWebsiteUrl": null,
        "requestMerchantProfileUpdate": {
          "id": "$UPDATE_REQUEST_ID",
          "status": "PendingReview",
          "merchantWebsiteUrl": "https://www.mygreatwebsite.fr/",
          "additionalInformation": [
            {
              "type": "NumberOfEmployees",
              "value": "5"
            }
          ]
        }
      }
    }
  }
}
```

## Step 4: Get the supporting documents collection {#step-4}

> Whenever supporting documents are required by a payment method, the merchant profile will have a supporting documents collection with the `WaitingForDocument` status.

Reviewer: ```suggestion
Whenever supporting documents are required by a payment method, the merchant profile will have a supporting documents collection field added to the `WaitingForDocument` status.
```

1. Use the `merchantProfile` query.
1. Add the `supportingDocumentCollections` field.
1. Optionally, use a filter to have only the collections waiting for document.
1. Add `requiredSupportingDocumentPurposes` with the fields `name` and `acceptableSupportingDocumentTypes` to know which documents need to be sent.

:::tip Document purposes
A purpose indicates what's needed (for example, `CompanyRegistration`). The `acceptableSupportingDocumentTypes` field lists which document types can satisfy each purpose.
:::

> 

Reviewer: Should we add a link to the Document center where we add more information about the types of documents collected?

---
title: Provide missing information
---

# Provide missing information

When your merchant payment method has the status `WaitingForInformation`, you need to provide missing verification requirements and supporting documents before the review can be finalized.

This guide shows you how to identify what's missing and provide it.

:::tip Webhook
Subscribe to the `MerchantPaymentMethod.Updated` webhook to know when the status of the payment method changes.
:::

## Guide {#guide}

1. See which information is missing by checking the verification requirements.
1. Check if your project requires additional information.
1. Update the merchant profile with the missing information.

> 1. Get the supporting documents collection.

Reviewer: Get the supporting documents collection ID - matches with Step 4.

> 1. Get the supporting documents collection.

Author: ```suggestion
1. Get the supporting documents collection ID.
```

            ... on DefaultAdditionalInformation {
              type
              value
            }
          }
        }
      }
    }
    ... on Rejection {
      message
    }
    ... on ValidationRejection {
      fields {
        path
        code
        message
      }
    }
  }
}
```

#### Payload {#payload-update}

The new provided values appear under `requestMerchantProfileUpdate`. The values will only be reflected on the merchant profile after the request is approved by Swan.

```json {} showLineNumbers
{
  "data": {
    "requestMerchantProfileUpdate": {
      "__typename": "RequestMerchantProfileUpdateSuccessPayload",
      "merchantProfile": {
        "id": "$MERCHANT_PROFILE_ID",
        "merchantWebsiteUrl": null,
        "requestMerchantProfileUpdate": {
          "id": "$UPDATE_REQUEST_ID",
          "status": "PendingReview",
          "merchantWebsiteUrl": "https://www.mygreatwebsite.fr/",
          "additionalInformation": [
            {
              "type": "NumberOfEmployees",
              "value": "5"
            }
          ]
        }
      }
    }
  }
}
```

### Step 4: Get the supporting documents collection ID {#step-4}

> When supporting documents are required by a payment method, the merchant profile has a supporting documents collection with the `WaitingForDocument` status.

Reviewer: Should this say "supporting documents collection ID"?
